studio.py

```diff
@@ -14,14 +29,201 @@
+            if isinstance(self.app, AbstractAndroidFirefox):
+
+                device = ADBAndroid()
+                self.information["ro.product.model"] = device.shell_output(
+                    "getprop ro.product.model"
+                )
+                self.information["ro.build.user"] = device.shell_output(
+                    "getprop ro.build.user"
+                )
+                self.information["ro.build.version.release"] = device.shell_output(
+                    "getprop ro.build.version.release"
+                )
+
+                platform_name = "".join(
+                    e for e in self.information["device_name"] if e.isalnum()
+                )
+
+            for site in sites_json:
+                label = site.get("label")
+                if label:
+                    name = "-".join(
+                        [
+                            self.proxy,
+                            platform_name,
+                            self.app.lower(),
+                            site["domain"],
+                            label,
+                        ]
+                    )
+                else:
+                    name = "-".join(
+                        [self.proxy, platform_name, self.app.lower(), site["domain"]]
+                    )
+                site["path"] = os.path.join(self.path, name, name)
+                site["name"] = name
+
+                site["recording_path"] = "%s.mp" % site["path"]
+                site["json_path"] = "%s.json" % site["path"]
+                site["screen_shot_path"] = "%s.png" % site["path"]
+                site["zip_path"] = os.path.join(self.path, "%s.zip" % site["name"])
+                site["manifest_path"] = os.path.join(
+                    self.path, "%s.manifest" % site["name"]
+                )
+
+                sites.append(site)
+        else:
+            raise Exception("No site file found!!!")
+
+        return sites
+
+    def update_json_information(self, site):
+        time.sleep(3)
+        print("Updating json with recording information")
+
+        with open(site["json_path"], "r") as f:
+            json_data = json.loads(f.read())
+
+        self.information["proxy"] = self.proxy
+
+        self.information["url"] = site["url"]
+        self.information["domain"] = site["domain"]
+
+        self.information["label"] = site.get("label")
+
+        json_data["info"] = self.information
+        with open(site["json_path"], "w") as f:
+            f.write(json.dumps(json_data))
+
+    def generate_zip_file(self, site):
+        print("Generating zip file")
+
+        with ZipFile(site["zip_path"], "w") as zf:
+
+            zf.write(site["recording_path"], os.path.basename(site["recording_path"]))
+            zf.write(site["json_path"], os.path.basename(site["json_path"]))
+            zf.write(
+                site["screen_shot_path"], os.path.basename(site["screen_shot_path"])
+            )
+
+    def generate_manifest_file(self, site):
+
```

Comment:
Nit: Remove blank line

Reply:
Can you run black over your changes, as it should take care of things like this and means we don't need to comment on them in reviews.